### ISSUE TITLE
Style/RedundantCondition: False positive

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#13964]
## Expected behavior

```
$ cat rubocop/redundant_condition.rb
def true?(x) = !!x
def true_or_nil(x) = true?(x) ? true : nil
```

The ternary is definitely not optional.  It could be a postfix `if`, granted.  But it is not wrong.

## Actual behavior

```
$ bundle exec rubocop rubocop/redundant_condition.rb
Inspecting 1 file
C

Offenses:

rubocop/redundant_condition.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
def true?(x) = !!x
^
rubocop/redundant_condition.rb:2:31: C: [Correctable] Style/RedundantCondition: Use double pipes || instead.
def true_or_nil(x) = true?(x) ? true : nil
                              ^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

This would break the code.

## Steps to reproduce the problem

See above.

## RuboCop version

```
$ bundle exec rubocop -V
1.73.2 (using Parser 3.3.7.1, rubocop-ast 1.38.1, analyzing as Ruby 3.3, running on ruby 3.3.7) [arm64-darwin24]
  - rubocop-minitest 0.37.1
  - rubocop-performance 1.24.0
```

Cheers!

**Comment by Earlopain** [comment]
Hi, am I missing something?

```rb
irb
irb(main):001> true ? true : nil
=> true
irb(main):002> false ? true : nil
=> nil
irb(main):003> true || nil
=> true
irb(main):004> false || nil
=> nil
```

**Comment by Earlopain** [comment]
Right, yeah .That makes sense. There is https://github.com/rubocop/rubocop/issues/13931 for this but it was closed as wont-fix, not making the cop unsafe. I would encourage you to write there if you ran into issues because of this in the real world.
